OutOfSchool/OutOfSchool.WebApi/Controllers/V1/AdminController.cs

```diff
@@ -9,36 +12,248 @@ namespace OutOfSchool.WebApi.Controllers.V1;
+       ISensitiveApplicationService applicationService,
+       ISensitiveDirectionService directionService,
+       ISensitiveProviderService providerService,
+       IStringLocalizer<SharedResource> localizer)
     {
-        this.userService = userService;
-        this.mapper = mapper;
+        this.localizer = localizer;
+        this.logger = logger;
+        this.applicationService = applicationService;
+        this.directionService = directionService;
+        this.providerService = providerService;
+        this.currentUserService = currentUserService ?? throw new ArgumentNullException(nameof(currentUserService));
+        this.ministryAdminService = ministryAdminService ?? throw new ArgumentNullException(nameof(ministryAdminService));
     }
 
     /// <summary>
-    /// To Get the Profile of authorized Admin.
+     /// Get MinistryAdmins that match filter's parameters.
+     /// </summary>
+     /// <param name="filter">Entity that represents searching parameters.</param>
+     /// <returns><see cref="SearchResult{MinistryAdminDto}"/>, or no content.</returns>
+     [HasPermission(Permissions.MinistryAdminRead)]
+     [HttpGet]
+     [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(SearchResult<MinistryAdminDto>))]
+     [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+     [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+     [ProducesResponseType(StatusCodes.Status403Forbidden)]
+     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+     public async Task<IActionResult> GetByFilterMinistryAdmin([FromQuery] MinistryAdminFilter filter)
+     {
+         if (!currentUserService.IsAdmin())
+         {
+             logger.LogError("You have no rights because you are not an admin");
+             return this.Forbid();
+         }
+
+         var ministryAdmins = await ministryAdminService.GetByFilter(filter).ConfigureAwait(false);
+
+         if (ministryAdmins.TotalAmount < 1)
+         {
+             return NoContent();
+         }
+
+         return Ok(ministryAdmins);
+     }
+
+     /// <summary>
+     /// Get all Providers from the database.
+     /// </summary>
+     /// <param name="filter">Filter to get a part of all providers that were found.</param>
+     /// <returns>The result is a <see cref="SearchResult{ProviderDto}"/> that contains the count of all found providers and a list of providers that were received.</returns>
+    [HasPermission(Permissions.ProviderRead)]
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(SearchResult<ProviderDto>))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<IActionResult> GetByFilterProvider([FromQuery] ProviderFilter filter)
```

Comment:
Thank you for the comment. We discussed this issue and found that these 2 methods really do practically the same logic, so we had a talk with front-end team and asked about their using these methods. They didn`t use ```GetProvider``` so it was removed

Reply:
Ok, I'll be more specific: there are `GetProviderByFilter` and `GetByFilterProvider`. Please check them as well

Reply:
> Ok, I'll be more specific: there are `GetProviderByFilter` and `GetByFilterProvider`. Please check them as well

I asked frontend team, they use as I understand- ```GetProviderByFilter```, so I remove the second one. Because they do the same job)